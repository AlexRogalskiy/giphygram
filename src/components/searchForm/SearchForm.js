import React from 'react';
import PropTypes from 'prop-types';

export class SearchForm extends React.Component {
  static propTypes = {
    onSearch: PropTypes.func,
  };

  static defaultProps = {
    onSearch: () => {},
  };

  constructor(props) {
    super(props);

    this.state = {
      query: '',
    };
  }

  onQueryChange = (event) => {
    const query = event.target.value;
    this.setState({query});
  };

  onFormSubmit = (event) => {
    event.preventDefault();

    const {query} = this.state;
    const {onSearch} = this.props;

    onSearch(query);
  };

  render() {
    const {query} = this.state;

    return (
      <form className="form-inline" onSubmit={this.onFormSubmit}>

        <div className="input-group">
          <input
            className="form-control"
            type="search"
            placeholder="Search GIFs"
            value={query}
            onChange={this.onQueryChange}
          />

          <div className="input-group-append">
            <button
              className="btn btn-outline-light"
              type="submit"
              onClick={this.onFormSubmit}>
              Search
            </button>
          </div>
        </div>
      </form>
    );
  }
}
