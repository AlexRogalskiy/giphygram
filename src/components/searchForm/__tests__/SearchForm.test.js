import React from 'react';
import renderer from 'react-test-renderer';
import { SearchForm } from '../SearchForm';

describe('SearchForm', () => {
  it('should be rendered correctly empty query', () => {
    const tree = renderer
      .create(<SearchForm/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be rendered correctly non-empty query', () => {
    const tree = renderer
      .create(<SearchForm query={'kittens'}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
