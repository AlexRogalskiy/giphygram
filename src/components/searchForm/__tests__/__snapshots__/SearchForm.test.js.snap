// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should be rendered correctly empty query 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="input-group"
  >
    <input
      className="form-control search-input"
      onChange={[Function]}
      placeholder="Search GIFs"
      type="search"
      value=""
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-dark"
        onClick={[Function]}
        type="submit"
      >
        <span
          className="oi oi-magnifying-glass"
        />
      </button>
    </div>
  </div>
</form>
`;

exports[`SearchForm should be rendered correctly non-empty query 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <div
    className="input-group"
  >
    <input
      className="form-control search-input"
      onChange={[Function]}
      placeholder="Search GIFs"
      type="search"
      value="kittens"
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-dark"
        onClick={[Function]}
        type="submit"
      >
        <span
          className="oi oi-magnifying-glass"
        />
      </button>
    </div>
  </div>
</form>
`;
